Sources/POSIX/realpath.swift

```diff
@@ -27,3 +27,31 @@ public func realpath(_ path: String) throws -> String {
     guard let rvv = String(validatingUTF8: rv!) else { throw SystemError.realpath(-1, path) }
     return rvv
 }
+
+private let pathComponentSeparator = "/"
+
+/**
+ Resolves executable, both absolute and relative paths and referred from `PATH` environment variable and
+ return the absolute pathname.
+
+ All components in executable must exists when realpath(executable:) is called.
+*/
+public func realpath(executable: String) throws -> String {
+    if executable.starts(with: pathComponentSeparator) {
```

Comment:
I usually use `hasPrefix` method instead of `starts(with:)`. I am not sure if performance of one is better than other or if it doesn't really matter. @airspeedswift Do you know?

Reply:
It maybe possible that starts(with:) is slower because it comes from Sequence but also the compiler could be optimizing it. Either way, I think hasPrefix may be more "Swifty".

Reply:
It's an anomaly that we have both, that we should fix by unifying. However, as of right now `hasPrefix` can be faster because it can use memcmp for ascii if the two strings are both ascii.

Reply:
Great, thanks for explaining!